### PR TITLE
Recreate table overviews on synchronizations

### PR DESCRIPTION
**Pull request by jgoizueta**
When a sync table had overviews created on the initial import, these persisted unmodified for ensuing synchronizations,
while the newly updated base table was lacking the _feature_count column. So we had two problems:
* Base table without _feature_count is inconsistence with the existence of overview tables
* Overviews tables where out of date with base table

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Development
 * An attacker could execute commands in the server running the queues by importing a file with a carefully crafted filename. Fixed in #11782
 
 ### Features
+* Overiviews are synchronized now (#12092)
```
Comment (ethervoid) [inline NEWS.md]: Typo s/Overiviews/Overviews

File: services/importer/lib/importer/overviews.rb [hunk 2]
```diff
@@ -79,6 +88,10 @@ def create_overviews!
           @service.create_overviews! @table_name
         end
 
+        def delete_overviews!
```
Comment (ethervoid) [inline services/importer/lib/importer/overviews.rb]: We could add here an `update_overviews` method that act as recreate

File: app/models/synchronization/adapter.rb [hunk 3]
```diff
@@ -48,6 +50,28 @@ def run(&tracker)
         raise exception
       end
 
+      def recreate_overviews(result)
```
Comment (ethervoid) [inline app/models/synchronization/adapter.rb]: We have two methods to deal with overviews, one in the importer and this one. 

Maybe we should move this logic to [the overviews class](https://github.com/CartoDB/cartodb/blob/master/services/importer/lib/importer/overviews.rb). There we could do exactly what we're doing here or in [here](https://github.com/CartoDB/cartodb/blob/master/app/connectors/importer.rb#L116-L128)

File: app/models/synchronization/adapter.rb [hunk 4]
```diff
@@ -36,6 +37,7 @@ def run(&tracker)
           overwrite(table_name, result)
           setup_table(table_name, geo_type)
           run_index_statements(index_statements)
+          recreate_overviews(result)
```
Comment (ethervoid) [inline app/models/synchronization/adapter.rb]: Here we could do: `@overviews_creator.dataset(result.name).update_overviews()` for example

File: NEWS.md (hunk 1 above)
Comment (jgoizueta) [inline NEWS.md]: Fixing it, thanks!

File: app/models/synchronization/adapter.rb (hunk 3 above)
Comment (jgoizueta) [inline app/models/synchronization/adapter.rb]: After having taken a look at this I also think we should move the overview-recreation logic out of here. But I think I'll keep this method here handling exceptions.

Comment (jgoizueta) [inline app/models/synchronization/adapter.rb]: I'm no longer sure if I prefer the current situation ir moving logic out of here! 😅 

Let's see:

The `Overviews` class takes care of  overviews maintenance logic, exposed through Overviews::Dataset.
(and using the low level overview operations encapsulated in `OverviewsService`)

Consumers of overviews are:
* Imports (`Connector::Importer` app/connectors/importer.rb)
* Syncs (`Synchroniztion::Adapter` app/models/synchronizations/adapter.rb)

And they should just request overview maintenance and handle exceptions.

### A. (current situation)

* `should_create_overviews?` information is independent of overview creation.

This allows adding logic in the consumers, such as additional logging, or aborting sync creation if `interval` is too short for overviews

### B. (proposed)

`Overviews` knows if overviews should be created so it handles this internally:

```ruby
def create_overviews
  @service.create_overviews! if should_create_overviews? 
end

def update_overviews
  delete_overviews!
  create_overviews
end
```

* Then we cannot easily do additional logging at the consumer.
* To allow sync limits we should continue exposing `should_create_overviews?` (probably with a better name: `will_have_overviews?`)

I'll hold off this decision, in hope that tomorrow there will be not so many bands doing soundchecks almost inside CARTO :guitar:. Feedback is welcome in the meantime.

File: spec/models/synchronization/member_overviews_spec.rb [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+require_relative '../../../services/data-repository/repository'
+require_relative '../../../app/models/synchronization/member'
+require 'helpers/unique_names_helper'
+require 'helpers/file_server_helper'
+require_relative '../../helpers/feature_flag_helper'
+
+include UniqueNamesHelper
+include CartoDB
+
+describe Synchronization::Member do
```
Comment (jgoizueta) [inline spec/models/synchronization/member_overviews_spec.rb]: Yep. Sure. Lots of lines.

**Comment by jgoizueta** [comment]
I've added some tests, fixed a problem and refactored a little bit the overviews-handling logic. @ethervoid: a little review?

**Comment by rochoa** [comment]
What is the status of this PR? I see changes were requested in the CR.

**Comment by ethervoid** [comment]
@rochoa My fault, he asked me for a review. I'll do it in a minute and we'll move this forward

**Comment by ethervoid** [comment]
@jgoizueta I've checked the changes and looks good to me, good catch with the bug

**Comment by rochoa** [comment]
🎉

**Pull request merged**